.travis.yml

```diff
@@ -1,17 +1,13 @@
 language: rust
-sudo: false
 cache: cargo
-before_script:
-- cargo install -f rustfmt
-- export PATH=$PATH:$HOME/.cargo/bin
 rust:
-- nightly
-- beta
-- stable
-matrix:
-  allow_failures:
-  - rust: nightly
+  - stable
+  - beta
+  - nightly
 script:
-- cargo fmt -- --write-mode=diff
-- cargo build --verbose
-- cargo test --verbose
+  - cargo test
+matrix:
+  include:
+    - rust: stable
+      before_script: rustup component add rustfmt-preview
+      script: cargo fmt --all -- --check
```

Comment:
Can you add a `env: RUN=FMT` here, that's easier to check on Travis if that's the fmt that fail.